### PR TITLE
upgrade to lucene 9

### PR DESCRIPTION
**Pull request by vthacker**
Lucene 9 is the next major version that released over a month ago. 

Right now our usage of lucene API is minimal so upgrading is much easier. Before we start working on more search improvements and use more of lucene APIs let's upgrade first 

complete list of changes are here - https://lucene.apache.org/core/9_0_0/changes/Changes.html

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/logstore/index/KalDBMergeScheduler.java [hunk 1]
```diff
@@ -31,11 +30,11 @@ public KalDBMergeScheduler(MeterRegistry metricsRegistry) {
     this.mergeCounter = this.metricsRegistry.counter(MERGE_COUNTER);
   }
 
-  protected void doMerge(IndexWriter writer, MergePolicy.OneMerge merge) throws IOException {
+  protected void doMerge(MergeSource mergeSource, MergePolicy.OneMerge merge) throws IOException {
```
Comment (vthacker) [inline kaldb/src/main/java/com/slack/kaldb/logstore/index/KalDBMergeScheduler.java:33]: https://issues.apache.org/jira/browse/LUCENE-9345

File: kaldb/pom.xml [hunk 2]
```diff
@@ -134,7 +134,7 @@
         </dependency>
         <dependency>
             <groupId>org.apache.lucene</groupId>
-            <artifactId>lucene-analyzers-common</artifactId>
+            <artifactId>lucene-analysis-common</artifactId>
```
Comment (vthacker) [inline kaldb/pom.xml:137]: https://issues.apache.org/jira/browse/LUCENE-9562

File: kaldb/src/test/java/com/slack/kaldb/chunk/IndexingChunkImplTest.java [hunk 3]
```diff
@@ -532,7 +532,7 @@ public void testSnapshotToS3UsingChunkApi() throws Exception {
       assertThat(chunk.snapshotToS3(bucket, "", s3BlobFs)).isTrue();
       assertThat(chunk.info().getSnapshotPath()).isNotEmpty();
 
-      assertThat(getCount(INDEX_FILES_UPLOAD, registry)).isEqualTo(15);
+      assertThat(getCount(INDEX_FILES_UPLOAD, registry)).isEqualTo(18);
```
Comment (vthacker) [inline kaldb/src/test/java/com/slack/kaldb/chunk/IndexingChunkImplTest.java:535]: seems super brittle! I just "fixed" it by changing to the new file number count that lucene 9 will produce

**Comment by mansu** [comment]
Let's hold off on the Lucene 9 upgrade until we better understand how we will handle field conflicts a bit better.

**Pull request closed without merging**